src/praisonai/tests/unit/test_approval_agent_integration.py

```
@@ -132,24 +131,28 @@ def auto_deny_callback(function_name, arguments, risk_level):
         print(f"❌ Auto-denial test failed: {e}")
-        return False
+        assert False, f"Auto-denial test failed: {e}"
 
 def test_agent_python_code_execution():
     """Test Python code execution through agent with approval."""
     print("\n🐍 Testing Agent Python Code Execution")
     print("=" * 45)
     
+    # Check if required packages are available - skip if not
+    try:
+        import black, pylint, autopep8
```

Ruff (0.11.9)

`black` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)